cdc/sink/mq/mq_flush_worker.go

```diff
@@ -235,17 +223,11 @@ func (w *flushWorker) run(ctx context.Context) (retErr error) {
 }
 
 // addEvent is used to add one event to the flushWorker.
-// It will return an ErrMQWorkerClosed if the flushWorker has exited.
 func (w *flushWorker) addEvent(ctx context.Context, event mqEvent) error {
 	select {
 	case <-ctx.Done():
 		return errors.Trace(ctx.Err())
-	case err, ok := <-w.errCh:
-		if !ok {
-			return cerror.ErrMQWorkerClosed.GenWithStackByArgs()
-		}
-		return err
-	case w.msgChan <- event:
+	case w.msgChan.In() <- event:
```

Comment:
It's possible, but it doesn't matter, the worker's run exits and no more messages are processed. And sink will be close soon.

Reply:
However, writing data to a closed chan will lead to panic.
Ref: https://github.com/pingcap/tiflow/pull/5675/files/87b4774faf66b3fcdc126b90f325e33b934a3b47#diff-28a85a85eb9cf5f4d9dd0de5f7e9a0d5ee84e1b2244c914d6ee48bd814705603R182

Reply:
I don't think there will be such a problem because the global sink close is the processor's responsibility, and the processor stops all table pipelines before shutting down the sink.